engine/engine_test.go

```diff
@@ -2985,6 +2985,42 @@ func TestQueryStats(t *testing.T) {
 	stats.NewQueryStats(q.Stats())
 }
 
+func TestQueryConcurrency(t *testing.T) {
```

Comment:
This is a small test to ensure there's no difference in the results of queries depending on the number of shards used.